src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py

```diff
@@ -273,7 +277,18 @@ def execute(self, emulator=None):
 
         os.environ.update(envOverride)
 
-        returnCode = subprocess.call(args, stdout=stdoutHandle, stderr=stderrHandle)
+        time0 = time.time()
+        # returnCode = subprocess.call(args, stdout=stdoutHandle, stderr=stderrHandle)
+        proc = psutil.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        cpuTime = 0
+        while proc.is_running():
```

Comment:
I fear like this might be too invasive, in terms of resources required for the monitoring. Giving that we are polling cpu_times every second, it is not clear to me whether it would properly account CPU allocation for this process. Did you check that?

I think the alternative I would suggest would be to append the `time` command to the subprocess args and use that for the wallclocktime (real) and cpu time (sys). But again, we need to first make sure that those metrics provide us with exactly what we need.

Reply:
it is resolved in new commit.